File: fsharp/AdventOfCode.UnitTests/Puzzles/Day07Should.fs
module AdventOfCode.UnitTests.Puzzles.Day07Should

open AdventureOfCode.Puzzles
open Xunit

[<Fact>]
let ``Return the correct calculation for the example of part 1`` () =
    let result = Day07.solvePart1 "Puzzles/Day07TestInput.txt"
    Assert.NotNull(result)
    Assert.Equal(37, result)
    
    
[<Fact>]
let ``Return the correct calculation for part 1`` () =
    let result = Day07.solvePart1 "Puzzles/Day07Input.txt"
    Assert.NotNull(result)
    Assert.Equal(356990, result)
    
[<Fact>]
let ``Return the correct calculation for the example of part 2`` () =
    let result = Day07.solvePart2 "Puzzles/Day07TestInput.txt"
    Assert.NotNull(result)
    Assert.Equal(168, result)
    
[<Fact>]
let ``Return the correct calculation for part 2`` () =
    let result = Day07.solvePart2 "Puzzles/Day07Input.txt"
    Assert.NotNull(result)
    Assert.Equal(101267361, result)
    

File: csharp/older/AdventOfCode.UnitTests/Puzzles/Day07Should.cs
using AdventOfCode.Core.Puzzles.Day07;
using Xunit;

namespace AdventOfCode.UnitTests.Puzzles;

public class Day07Should
{
    [Theory]
    [InlineData("TestInput/day07.txt", 37)]
    [InlineData("Input/day07.txt", 356990)]
    public void SolvePart1(string inputFile, int expectedResult)
    {
        // ARRANGE
        Solution day7 = new (inputFile);

        // ACT
        var result = day7.SolvePart1();

        // ASSERT
        Assert.NotNull(result);
        Assert.Equal(expectedResult, result);
    }
    
    [Theory]
    [InlineData("TestInput/day07.txt", 168)]
    [InlineData("Input/day07.txt", 101267361)]
    public void SolvePart2(string inputFile, int expectedResult)
    {
        // ARRANGE
        Solution day7 = new (inputFile);

        // ACT
        var result = day7.SolvePart2();

        // ASSERT
        Assert.NotNull(result);
        Assert.Equal(expectedResult, result);
    }
}